src/plugin/kubearmor.go

```diff
@@ -407,7 +404,9 @@ func StartKubeArmorRelay(StopChan chan struct{}, cfg types.ConfigKubeArmorRelay)
 		defer func() {
 			log.Info().Msg("watchlogs returning")
 			KubeArmorRelayStarted = false
-			_ = conn.Close()
+			if conn != nil {
+				_ = conn.Close()
+			}
```

Comment:
this won't be required as we are already checking for nil pointer above: line 389